src/data/multiset/basic.lean

```diff
@@ -37,6 +37,17 @@ instance has_decidable_eq [decidable_eq α] : decidable_eq (multiset α)
 | s₁ s₂ := quotient.rec_on_subsingleton₂ s₁ s₂ $ λ l₁ l₂,
   decidable_of_iff' _ quotient.eq
 
+private def sizeof_core [has_sizeof α] (s : multiset α) : ℕ :=
+begin
+  induction s with l x y ih, { exact l.sizeof },
+  simp only [eq_rec_constant],
+  induction ih with hd l₁ l₂ h₁₂ h_sz₁₂ a b l l₁ l₂ l₃ h₁₂ h₂₃ h_sz₁₂ h_sz₂₃,
+  { refl }, { simp [list.sizeof, h_sz₁₂] },
+  { dsimp [list.sizeof], apply add_left_comm }, { simp [h_sz₁₂, h_sz₂₃] }
+end
+
+instance has_sizeof [has_sizeof α] : has_sizeof (multiset α) := ⟨sizeof_core⟩
```

Comment:
If you put `@[priority 2000]` (higher than the default `has_sizeof` instance), do you still need to write your own instance for `finset`?

Reply:
The default instance says. `finset.rec (lam val nodup, 1 + sizeof val + sizeof nodup)`. So in principle the proof should go through... I'll try to make the proof work without the instance. (I couldn't unfold the definition previously).

The other instance (for multiset is definitely necessary, though, because) otherwise it falls back to `default.sizeof` which is just the constant `0` function.

Reply:
So, I've adjusted the proof to no longer need the instance for `finset`, good.